### ISSUE TITLE
Support for ingress-appgw add-on in azurerm_kubernetes_cluster

### ISSUE THREAD

**Issue by tkubica12** [terraform-providers/terraform-provider-azurerm#7384]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AKS ability to deploy managed Ingress controller for Azure Application Gateway is now in preview as add-on. [https://docs.microsoft.com/en-us/azure/application-gateway/ingress-controller-overview](https://docs.microsoft.com/en-us/azure/application-gateway/ingress-controller-overview)

Would you consider adding to azurerm_kubernetes_cluster, please?

### New or Affected Resource(s)

* azurerm_XXXXX

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by dhananjaya-senanayake** [comment]
@tombuildsstuff, 
Any update on this?
A timeline would be much appreciated.

**Comment by swies0wl** [comment]
App Gateway is now GA and no longer preview.  Can this be revisited and implemented?
- https://azure.microsoft.com/en-us/updates/general-availability-app-gateway-ingress-controller-addon-for-aks/
- https://docs.microsoft.com/en-us/azure/application-gateway/tutorial-ingress-controller-add-on-existing

**Comment by mazen423** [comment]
That should be considered as it's highly used

**Comment by JamesTaylorEsendex** [comment]
The preview label should be removed now that this is GA.

**Comment by jkroepke** [comment]
@katbyte any timeline to see this PR in a stable version? Thanks.

**Comment by JamesTaylorEsendex** [comment]
> @katbyte any timeline to see this PR in a stable version? Thanks.

It's available as of 2.57.0.
